app/assets/javascripts/discourse/app/templates/components/pick-files-button.hbs

```diff
@@ -1,6 +1,10 @@
-{{d-button action=(action "openSystemFilePicker") label=label icon=icon}}
-{{#if acceptAnyFile}}
-  <input type="file">
-{{else}}
-  <input type="file" accept={{acceptedFileTypesString}}>
+{{#if allowUpload}}
```

Comment:
It feels strange for this to be in the component because `allowUpload` has to be `true` for the component to even be useful. If that is the case, I think the component that is rendering this component should just avoid rendering it instead.

Reply:
That's fair enough, will extract it out!